Sage/carbon
# refactor(label): convert files to typescript FE-4861

### Proposed behaviour


Convert `Label` component files to TS

### Current behaviour


`Label` component is JS

### Checklist




#### QA


### Additional context



### Testing instructions


Test for regressions in inputs that use label

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


https://codesandbox.io/s/carbon-quickstart-typescript-forked-7xpo14?file=/src/index.tsx
Will likely need to export the project locally and confirm no types are broken
